Tasks/google-play-release/GooglePlay.ts

```diff
@@ -149,39 +87,34 @@ async function run() {
         // #7) Commit the edit transaction
 
         tl.debug(`Getting a package name from ${apkFile}`);
-        let packageName: string = manifest.package;
-        updateGlobalParams(globalParams, 'packageName', packageName);
+        const packageName: string = manifest.package;
+        googleutil.updateGlobalParams(globalParams, 'packageName', packageName);
 
         tl.debug('Initializing JWT.');
-        let jwtClient: any = new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
+        const jwtClient: any = googleutil.getJWT(key);
         globalParams.auth = jwtClient;
 
         tl.debug('Initializing Google Play publisher API.');
-        let edits: any = publisher.edits;
-        [edits, edits.apklistings, edits.apks, edits.tracks, edits.listings, edits.images, jwtClient].forEach(bb.promisifyAll);
+        const edits: any = googleutil.publisher.edits;
 
         tl.debug('Authorize JWT.');
-        await jwtClient.authorizeAsync();
+        await jwtClient.authorize();
 
         console.log(tl.loc('GetNewEditAfterAuth'));
         tl.debug('Creating a new edit transaction in Google Play.');
-        let currentEdit: Edit = await getNewEdit(edits, globalParams, packageName);
-        updateGlobalParams(globalParams, 'editId', currentEdit.id);
+        await googleutil.getNewEdit(edits, globalParams, packageName);
 
+        let requireTrackUpdate = false;
         if (shouldUploadApks) {
             tl.debug(`Uploading ${apkFileList.length} APK(s).`);
 
-            for (let apkFile of apkFileList) {
+            requireTrackUpdate = true;
+            for (const apkFile of apkFileList) {
                 tl.debug(`Uploading APK ${apkFile}`);
-                let apk: Apk = await addApk(edits, packageName, apkFile, APK_MIME_TYPE);
+                const apk: googleutil.Apk = await googleutil.addApk(edits, packageName, apkFile);
                 tl.debug(`Uploaded ${apkFile} with the version code ${apk.versionCode}`);
                 apkVersionCodes.push(apk.versionCode);
             }
-
-            console.log(tl.loc('UpdateTrack'));
```

Comment:
Just want to confirm that removing the `updateTrack` call is intentional?

Reply:
Basically, the `update` call is always made before; now it only make this call when there is metadata or changelog change.  Want to confirm that's the intended behavior.

Reply:
Hrmm... IIRC, I added instances where `updateTrack` is called.  I don't think I removed any.  Previously, I called during the condition that the APK was uploaded.  Now I additionally call when there is metadata or release notes have been attached.  I'm not aware of an instance where I removed a call.  If the case isn't covered by the variable `requireTrackUpdate`, then please let me know as it's probably a bug.

Reply:
In original file line 183, we are calling `updateTrack` if `shouldUploadApks` is true.  In the new flow, we make this call if `shouldAttachmetadata` (which is a user input) is true or if `changeLogFile` has content.  So there could be cases where we call `updateTrack` before, but not in the new flow.  

I don't know the API behavior, so it is certainly possible that we don't need to make `updateTrack` calls in those situations.

Reply:
For posterity, we spoke offline, and the case is covered by `requireTrackUpdate` in line 111